src/commonNative/kotlin/com/epam/drill/agent/State.kt

```diff
@@ -18,6 +19,21 @@ var isAsyncApp: Boolean
         _isAsyncApp.value = value.freeze()
     }
 
+private val _classScanDelay = AtomicReference(0L).freeze()
+
+var classScanDelay: Long
+    get() = _classScanDelay.value
+    set(value) {
+        _classScanDelay.value = value.freeze()
+    }
+
+val _latch = AtomicReference<Job?>(null).freeze()
```

Comment:
This latch just complicated the code. The delay can be implemented inside "getClassesByConfig" callback.